contracts/CollectiveGovernance.sol

```diff
@@ -100,27 +111,27 @@ contract CollectiveGovernance is Governance, VoteStrategy, ERC165 {
     }
 
     modifier requireVoteReady(uint256 _proposalId) {
-        require(_storage.isFinal(_proposalId), "Voting is not ready");
+        if(!(_storage.isFinal(_proposalId)) revert CG__VotingNotReady();
         _;
     }
 
     modifier requireVoteClosed(uint256 _proposalId) {
-        require(_storage.isFinal(_proposalId) && !isVoteOpenByProposalId[_proposalId], "Vote is not closed");
+        if(_storage.isFinal(_proposalId) && isVoteOpenByProposalId[_proposalId]) revert CG__VotingOpen();
```

Comment:
WIll need to double check these logic statements.  Would be good to make sure there is test coverage on it.   

This requires using De Morgan's Theorem quite a bit here.   I think it should be:
!isFinal || isVoteOpen

`!(A && B) == !A || !B`